consume-more-stuff/src/app/services/backend.service.ts

```diff
@@ -1,69 +1,75 @@
+import { HttpClient } from "@angular/common/http";
 
 @Injectable({
-    providedIn: 'root'
+    providedIn: "root"
 })
 
 export class BackendService {
-    baseUrl: string = "https://localhost:8989";
+    baseUrl: string = 'http://localhost:4200/'
     item: any[] = [];
     items: any
     obj: any;
-    id: any;
-    
+    id: number;
+ user:{
+     id:number,
+     username:string,
+    password: string,
+    name:string,
+    email:string,
+    user_status:number,
+    item_id: null
+    } = {
+     id:null,
+     username:'',
+     password:'',
+     name:'',
+     email:'',
+     user_status:null,
+     item_id: null,
+    };
     constructor(private http: HttpClient) { }
 
-    getAllUsers() {
-        // console.log(this.allUsers)
-        const url = this.baseUrl + '/users';
-        return this.http.get(url).toPromise();
-    }
-
     getUser(id: number) {
-        // console.log(this.users)
-        const url = this.baseUrl + '/users/' + id;
+        console.log('this is getting user')
+        const url = this.baseUrl + "/users/" + id;
         return this.http.get(url).toPromise();
     }
 
     getAllCategories() {
-        // console.log('this.allContacts in backend.service called\n', this.allContacts)
-        const url = this.baseUrl + '/categories';
+        const url = this.baseUrl + "api/categories";
         return this.http.get(url).toPromise();
     }
-
+    //can talk to database
     getAllItems() {
-        // console.log(this.allUsers)
-        const url = this.baseUrl + '/items';
+        console.log('getting all items')
+        const url = this.baseUrl + 'api/items';
         return this.http.get(url).toPromise();
     }
-
+    //can talk to database
     getItem(id: number) {
-        // console.log(this.users)
-        const url = this.baseUrl + '/items' + id;
-        return this.http.get(url).toPromise();
+        console.log('Getting Item')
+        const ItemUrl = this.baseUrl + 'api/items/' + `?id=${id}`
+        console.log('thisisid', id) 
+        return this.http.get(ItemUrl).toPromise()
     }
-
+                                   
     createItem(items) {
         console.log('items', items)
         this.item.push(items);
     }
 
     createNewItem(obj) {
-        const url = this.baseUrl + '/new';
-        console.log('createNewContact obj from backend.service', obj)
-        return this.http.post(url, obj)
-            .subscribe(res => {
-                console.log('added to DB')
-            });
+        const url = this.baseUrl + "/new";
+        console.log("createNewContact obj from backend.service", obj);
+        return this.http.post(url, obj).subscribe(res => {
+            console.log("added to DB");
+        });
     }
 
-    login(user) {
-        return Promise.resolve({
-            id: user.id,
-            username: user.username
-        });
+    login(username, password) {
+        const userUrl = this.baseUrl + `api/login/${username}`;
```

Comment:
To correspond with the route change to not include the username in the URL, you can remove the `${username}` here. The form will work the same.